STACpopulator/input.py

```diff
@@ -58,21 +63,47 @@ def reset(self):
         """Reset the generator."""
         self.catalog_head = self.catalog
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator[Tuple[str, MutableMapping[str, Any]]]:
         """Return a generator walking a THREDDS data catalog for datasets."""
-        yield from self.catalog_head.datasets.items()
+        if self.catalog_head.datasets.items():
+            for item_name, ds in self.catalog_head.datasets.items():
+                attrs = self.extract_metadata(ds)
+                yield item_name, attrs
 
         if self._depth > 0:
             for name, ref in self.catalog_head.catalog_refs.items():
                 self.catalog_head = ref.follow()
                 self._depth -= 1
                 yield from self
 
+    def extract_metadata(self, ds: siphon.catalog.Dataset) -> MutableMapping[str, Any]:
+        # Get URL for NCML service
+        url = ds.access_urls["NCML"]
+
+        LOGGER.info("Requesting NcML dataset description")
+        r = requests.get(url)
+
+        # Write response to temporary file
+        f = NamedTemporaryFile()
+        f.write(r.content)
+
+        # Convert NcML to CF-compliant dictionary
+        attrs = xncml.Dataset(f.name).to_cf_dict()
```

Comment:
Note to myself: I could add a `from_xml` class method to `xncml.Dataset` to avoid having to create a temp file.

Reply:
Ok, I'll make a PR for this.

Reply:
https://github.com/xarray-contrib/xncml/pull/54